### ISSUE TITLE
Warnings verbosity

### ISSUE THREAD

**Issue by joeflack4** [mapping-commons/sssom-py#500]
## Overview
I [noticed recently](https://github.com/monarch-initiative/mondo-ingest/pull/412#discussion_r1493469285) that sometimes warnings are overly verbose. For example, sometimes for a given type of warning, it prints a full warning for each instance of an offending class or mapping, rather than printing some text about the type of warning and then providing a concise list of all the offending cases.

## Current behavior
<details><summary>Example</summary>
<p>

In [my instance](https://github.com/monarch-initiative/mondo-ingest/pull/412#discussion_r1493469285), this warning repeated 100's of times, each time with a different mapping object, but the same error message. Here's what it looks like with just 3 entries (the latter 2 are fake / just examples).
```
WARNING:sssom.parsers:One mapping in the mapping set is not well-formed, and therfore not included in the mapping set ({'mapping_justification': 'semapv:UnspecifiedMatching', 'subject_id': 'Orphanet:3080', 'predicate_id': 'oboInOwl:hasDbXref', 'subject_label': 'Intellectual disability, Wolff type'}). Error: object_id must be supplied
WARNING:sssom.parsers:One mapping in the mapping set is not well-formed, and therfore not included in the mapping set ({'mapping_justification': 'semapv:UnspecifiedMatching', 'subject_id': 'Orphanet:3081', 'predicate_id': 'oboInOwl:hasDbXref', 'subject_label': 'LABEL'}). Error: object_id must be supplied
WARNING:sssom.parsers:One mapping in the mapping set is not well-formed, and therfore not included in the mapping set ({'mapping_justification': 'semapv:UnspecifiedMatching', 'subject_id': 'Orphanet:3082', 'predicate_id': 'oboInOwl:hasDbXref', 'subject_label': 'LABEL'}). Error: object_id must be supplied
```

</p>
</details> 

## Suggested behavior
<details><summary>Example</summary>
<p>

```
WARNING:sssom.parsers:Non well-formed mappings detected in mapping set, and therefore were excluded.
Error: object_id must be supplied
Offending mappings:
({'mapping_justification': 'semapv:UnspecifiedMatching', 'subject_id': 'Orphanet:3080', 'predicate_id': 'oboInOwl:hasDbXref', 'subject_label': 'Intellectual disability, Wolff type'}). 
({'mapping_justification': 'semapv:UnspecifiedMatching', 'subject_id': 'Orphanet:3081', 'predicate_id': 'oboInOwl:hasDbXref', 'subject_label': 'LABEL'}). 
({'mapping_justification': 'semapv:UnspecifiedMatching', 'subject_id': 'Orphanet:3082', 'predicate_id': 'oboInOwl:hasDbXref', 'subject_label': 'LABEL'}). 
```

</p>
</details> 

## Additional details
Also there is a typo, `therfore` in `WARNING:sssom.parsers:One mapping in the mapping set is not well-formed, and therfore not included in the mapping set...`.

**Comment by matentzn** [comment]
I agree with this, but isnt this going to be print the same number of rows? 

What about truncating the output and only print the first 10 cases, and then saying "There are 1230 similar ones, omitted for brevity"?

**Comment by matentzn** [comment]
I explored catching the warnings and basically it will require a huge amount of refactoring, because the warning is written a few methods down from where I could collect the erroneous mappings, that that method is called in dozens of places.

in #502 I have 
1. Drastically reduced the verbosity of the error your are particularly concerned with
1. Devised a generic strategy for truncating logs when in a low log level is set (WARNING or below) by extending the root logger and customising it. What it does, in essence, is documenting the source of a logging attempt. If that error has been printed more time than the limit (configurable), it just does not print it. In the end of a command execution, a log summary is printed if and only if an error was truncated.

What do you think?

**Comment by joeflack4** [comment]
Wow you got right to this. Yeah I think this seems pretty good!
